### PR TITLE
[gsoc2020] Merged Modules remove django-netjsongraph

### PR DESCRIPTION
**Pull request by atb00ker**
Blocked by `openwisp-network-topology release` after https://github.com/openwisp/openwisp-network-topology/issues/66

### CODE REVIEW

File: tasks/pip.yml [hunk 1]
```diff
@@ -124,14 +122,13 @@
...
-  when: openwisp2_python_packages
+  when: openwisp2_python_packages|length > 0
   pip:
     name: "{{ openwisp2_python_packages }}"
-    state: latest
+    state: forcereinstall
     virtualenv: "{{ virtualenv_path }}"
     virtualenv_python: "{{ openwisp2_python }}"
     virtualenv_site_packages: yes
-    extra_args: "--no-deps"
```
[the remark this answers is not in the document]
Comment (nemesisdesign) [reply tasks/pip.yml:134]: when using custom targets (eg: the master branches) I found that the system would download and install the dependencies from pypi, overriding the custom dependencies just downloaded.

Eg:

- openwisp-utils 0.6.0 alpha is installed from github
- installing openwisp-controller master from github downloads and installs openwisp-utils 0.5.0 stable, hence losing openwisp-utils 0.6.0. I haven't found a solution to this yet, but if you do (it's quite likely it can be done since I didn't have a lot of time to dedicate to this) please let me know and maybe we can solve another painful point here

[the remark this answers is not in the document]
Comment (atb00ker) [reply tasks/pip.yml:134]: I am not able to replicate it,
I did:
```
openwisp2_utils_pip: https://github.com/atb00ker/openwisp-utils/tarball/master
openwisp2_controller_pip: https://github.com/openwisp/openwisp-controller/tarball/master
```

Where I randomly put util's version as `0.9.0` in my fork's master branch and I am seeing `0.9.0` only.

[the remark this answers is not in the document]
Comment (nemesisdesign) [reply tasks/pip.yml:134]: ok, let's remove it, I'll try an alternative and cleaner solution

File: tasks/pip.yml [hunk 2]
```diff
@@ -113,7 +112,7 @@
 - name: Install extra python packages
   pip:
     name: "{{ openwisp2_extra_python_packages }}"
-    state: latest
+    state: forcereinstall
```
Comment (nemesisdesign) [inline tasks/pip.yml:115]: why is this needed?

File: tasks/pip.yml [hunk 3]
```diff
@@ -122,21 +121,24 @@
   register: result
   until: result is success
   notify: reload supervisor
+  tags:
+    - molecule-idempotence-notest
```
Comment (nemesisdesign) [inline tasks/pip.yml:125]: why is this needed?

File: tasks/pip.yml [hunk 4]
```diff
@@ -122,21 +121,24 @@
...
     virtualenv_python: "{{ openwisp2_python }}"
     virtualenv_site_packages: yes
-    extra_args: "--no-deps"
   notify: reload supervisor
   retries: 5
   delay: 10
   register: result
   until: result is success
+  tags:
+    - molecule-idempotence-notest
```
Comment (nemesisdesign) [inline tasks/pip.yml:141]: why is this needed?

File: tasks/pip.yml (hunk 2 above)
Comment (atb00ker) [reply tasks/pip.yml]: Before `openwisp-network-topology` was released, I was getting an error swapper isn't installed because of `--no-deps` didn't allow dependencies to be installed. Hence, I removed `--no-deps`, then I realized that I was trying to install the same version of network-topology, but that wouldn't happen because it is already in `latest` (as version had not changed); I realized that a user may find themselves in the same problem trying to install their own version, so I changed it to `forceinstall`, which causes idempotence to fail, so I had to add `molecule-idempotence-notest`.

But after the release I am not able to reliably reproduce this problem with openwisp-utils. Hence, I am undoing all these changes. It's not specifically related to this PR and will stall this work needlessly.

But I think we should open a separate issue for this `--no-deps` problem.
Please let me know what do you think about it. :smile:

Comment (nemesisdesign) [reply tasks/pip.yml]: @atb00ker yes ok, let's open another issue for this :+1:
